addons/gamification/models/challenge.py

```diff
@@ -328,6 +328,7 @@ def _generate_goals_from_challenge(self):
         can be called after each change in the list of users or lines.
         :param list(int) ids: the list of challenge concerned"""
 
+        commit = self.env.context.get('commit_gamification') and self.env.cr.commit
```

Comment:
move the if into the loop directly
key = 'auto_commit' / 'force_commit'

I don't see anywhere where you are using this context. Cron should use it.